pkg/virt-controller/watch/volume-migration/volume-migration.go

```diff
@@ -141,6 +146,34 @@ func ValidateVolumes(vmi *virtv1.VirtualMachineInstance, vm *virtv1.VirtualMachi
 			valid = false
 			continue
 		}
+
+		// DataVolumes with a no-csi storage class
+		if v.VolumeSource.DataVolume != nil {
+			dv, err := storagetypes.GetDataVolumeFromCache(vm.Namespace, v.VolumeSource.DataVolume.Name, dvStore)
+			if err != nil {
+				return err
+			}
+			if dv == nil {
+				return fmt.Errorf("the datavolume %s doesn't exist", v.VolumeSource.DataVolume.Name)
```

Comment:
Perhaps I'm being pedantic, but is this really an error? Or something we should handle in eventually consistent system?

At the end of the day it may not matter much, but the VM controller is watching DataVolumes so if we returned some indication here to the caller to stop reconciling for now (instead of an error), the controller will reconcile again when the DV is created

Reply:
@mhenriks how would handled it? If I don't return an error then the function won't be re-evaluated, and until the dv isn't created then we cannot really validate the volume. So, from the point of view of the function, this is an error.

Reply:
@mhenriks do you mind if we merge as it is, and eventually, we will fix this later?